src/base_client.ts

```diff
@@ -82,8 +82,18 @@ export default class BaseClient {
             reject(new Error(`Unsupported Redirect Response: ${statusCode}`));
           }
           else if (statusCode >= 400 && statusCode <= 599) {
-            const errJson = JSON.parse(body);
             const { statusCode, headers } = response;
+
+            let errJson;
+            try {
+              errJson = JSON.parse(body);
+            } catch (_) {
+              return reject(new ErrorResponse({
+                error: 'Something went wrong, but could not parse the response',
```

Comment:
Ignore the but. You don't need but.

Reply:
Sometimes including the body itself could be revealing.

Reply:
I wanted to convey that bad something happened and separately we couldn't parse the response. Though I accept that perhaps the distinction is irrelevant to the user